pennylane/pauli/pauli_arithmetic.py

```diff
@@ -0,0 +1,279 @@
+
+        for wire in wire_order[1:]:
+            result_matrix = kron(result_matrix, matrix_map[self[wire]])
+
+        return result_matrix
+
+
+class PauliSentence(dict):
+    """Dict representing a Pauli Sentence. The keys are
+    PauliWord instances and the values correspond to coefficients.
+
+    >>> ps = PauliSentence({
+            PauliWord({0:X, 1:Y}): 1.23
+            PauliWord({2:Z, 0:Y}): -0.45j
+        })
+    """
+
+    def __missing__(self, key):
+        """If the PauliWord is not in the sentence then the coefficient
+        associated with it should be 0."""
+        return 0.0
+
+    def __add__(self, other):
+        """Add two Pauli sentence together by iterating over the smaller
+        one and adding its terms to the larger one."""
+        c_self, c_other = (copy(self), copy(other))
+        smaller_ps, larger_ps = (c_self, c_other) if len(self) < len(other) else (c_other, c_self)
+        for key in smaller_ps:
+            larger_ps[key] += smaller_ps[key]
+
+        return larger_ps
+
+    def __mul__(self, other):
+        """Multiply two Pauli sentences by iterating over each sentence and multiplying
+        the Pauli words pair-wise"""
+        final_ps = PauliSentence()
+
+        for pw1 in self:
+            for pw2 in other:
+                prod_pw, coeff = pw1 * pw2
+                final_ps[prod_pw] += coeff * self[pw1] * other[pw2]
+
+        return final_ps
+
+    def __str__(self):
+        """String representation of the PauliSentence."""
+        rep_str = ""
+        for index_i, (pw, coeff) in enumerate(self.items()):
+            if index_i == 0:
+                rep_str += "= "
+            else:
+                rep_str += "+ "
+            rep_str += f"({coeff}) * "
+
+            if len(pw) == 0:
+                rep_str += "[()]\n"
+                continue
+
+            for index_j, (w, op) in enumerate(pw.items()):
+                if index_j == 0:
+                    rep_str += f"[{op}({w})]"
+                else:
+                    rep_str += f" @ [{op}({w})]"
+            rep_str += "\n"
+
+        return rep_str
+
+    @property
+    def wires(self):
+        """Track wires of the PauliSentence."""
+        return set().union(*(pw.wires for pw in self.keys()))
+
+    def to_mat(self, wire_order, format="dense"):
+        """Given a Pauli word, get the matrix representation.
+
+        KeywordArgs:
+            wire_order (iterable or None): The order of qubits in the tensor product.
+            format (str): The format of the matrix ("dense" by default), if not a dense
+                matrix, then the format for the sparse representation of the matrix.
+
+        Returns:
+            (Union[NumpyArray, ScipySparseArray]): Matrix representation of the PauliSentence.
+
+        Rasies:
+            ValueError: Can't get the matrix of an empty PauliSentence.
+        """
+        if len(self) == 0:
+            raise ValueError("Can't get the matrix of an empty PauliSentence.")
```

Comment:
Identity?

Reply:
An empty `PauliSentence` and `PauliWord` act kind of like the identity on all wires. So I would be hesitant to return a fixed dimension matrix here since the behaviours wouldn't match. I think having a error is fine 👍🏼